File: custom_dir/custom_actions/restartgame.py
# -*- coding: UTF-8 -*-
from time import sleep
from maa.context import Context
from maa.custom_action import CustomAction
from maa.agent.agent_server import AgentServer
from maa.define import Status


@AgentServer.custom_action("ReStartGame")
class ReStartGame(CustomAction):
    cont=0
    def run(self, context: Context, argv: CustomAction.RunArg) -> bool:

        if self.cont <5:
            print("即将重启游戏并跳过任务")
            r=context.run_task("关闭阴阳师")
            if r==Status.failed:
                print("关闭游戏失败,检测游戏区服是否正确")
                return False
            sleep(3)
            context.run_task("启动游戏")
            print(f"重启完成,跳过任务")
            self.cont=self.cont+1
            print(f"当前重启次数为:{str(self.cont)}")
            # context.run_task("StopTask")
            # print(f"结束任务链")
            return True
        else:
            context.run_task("StopTask")
            return True
    def stop(self) -> None:
        pass

File: agent/actions/restartgame/restartgame.go
package restartgame

import (
	"fmt"
	"sync"
	"time"

	"github.com/MaaXYZ/maa-framework-go/v4"
)

type ReStartGame struct {
	count int
	mu    sync.Mutex
}

// Run 重启游戏
func (a *ReStartGame) Run(ctx *maa.Context, arg *maa.CustomActionArg) bool {
	a.mu.Lock()
	count := a.count
	a.mu.Unlock()

	if count < 5 {
		fmt.Println("即将重启游戏并跳过任务")
		r, err := ctx.RunTask("关闭阴阳师")
		if err != nil || r == nil {
			fmt.Printf("关闭游戏失败: %v, 检测游戏区服是否正确\n", err)
			return false
		}

		time.Sleep(3 * time.Second)
		_, _ = ctx.RunTask("启动游戏")
		fmt.Println("重启完成,跳过任务")

		a.mu.Lock()
		a.count++
		count = a.count
		a.mu.Unlock()

		fmt.Printf("当前重启次数为: %d\n", count)
		return true
	}

	_, _ = ctx.RunTask("StopTask")
	return true
}